llvm/lib/Target/PowerPC/PPCTargetTransformInfo.cpp

```diff
@@ -802,12 +802,18 @@ InstructionCost PPCTTIImpl::getMemoryOpCost(unsigned Opcode, Type *Src,
   // explicitly check this case. There are also corresponding store
   // instructions.
   unsigned MemBytes = Src->getPrimitiveSizeInBits();
-  if (ST->hasVSX() && IsAltivecType &&
-      (MemBytes == 64 || (ST->hasP8Vector() && MemBytes == 32)))
-    return 1;
+  unsigned SrcBytes = LT.second.getStoreSize();
+  if (ST->hasVSX() && IsAltivecType) {
+    if (MemBytes == 64 || (ST->hasP8Vector() && MemBytes == 32))
```

Comment:
nit: this maybe not related to the patch,

I think the variable `MemBytes` should be `MemBits` , otherwise it is not easy to understand (it looks like compare with 64bytes and 32bytes)

`if (MemBytes == 64 || (ST->hasP8Vector() && MemBytes == 32))`

I think we can modify the variable name in the patch?